src/main/java/com/polezhaiev/shop/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+        int total = 0;
+
+        ShoppingCart shoppingCart = shoppingCartRepository.getById(userId);
+        order.setUser(shoppingCart.getUser());
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        order.setStatus(Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setTotal(BigDecimal.ONE);
+        orderRepository.save(order);
+
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        Set<OrderItem> orderItems = new HashSet<>();
+        for (CartItem cartItem: cartItems) {
+            OrderItem tempOrderItem = new OrderItem();
+            tempOrderItem.setOrder(order);
+            tempOrderItem.setBook(cartItem.getBook());
+            tempOrderItem.setPrice(cartItem.getBook().getPrice());
+            tempOrderItem.setQuantity(cartItem.getQuantity());
+            orderItems.add(tempOrderItem);
+
+            total += tempOrderItem.getQuantity().intValue()
+                    * tempOrderItem.getPrice().intValue();
+
+            orderItemRepository.save(tempOrderItem);
+        }
+        order.setOrderItems(orderItems);
+        order.setTotal(BigDecimal.valueOf(total));
+        Order savedOrder = orderRepository.save(order);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public List<OrderResponseDto> findAllOrders(Long userId) {
+        return orderRepository.findAllByUserId(userId)
+                .stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemResponseDto> findAllItemsInOrder(Long orderId) {
+        return orderRepository.getById(orderId).getOrderItems()
+                .stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto findItemForOrder(Long orderId, Long orderItemId) {
+        return orderRepository.getById(orderId).getOrderItems()
+                .stream()
+                .filter(i -> i.getId().equals(orderItemId))
+                .map(orderItemMapper::toDto)
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "There is no item for order by id: " + orderId)
+                );
+    }
+
+    @Override
+    public OrderResponseDto updateStatus(Long orderId, OrderUpdateStatusRequestDto requestDto) {
+        Order order = orderRepository.getById(orderId);
+        order.setStatus(requestDto.getStatus());
+        Order updated = orderRepository.save(order);
+        return orderMapper.toDto(updated);
+    }
+
+    @Override
+    public void deleteOrderItemById(Long userId, Long id) {
+        Long userIdFromRepo = orderItemRepository.getById(id).getOrder().getUser().getId();
+        if (userId.equals(userIdFromRepo)) {
+            orderItemRepository.deleteById(id);
+        } else {
+            throw new NoSuchElementException(

```

Comment:
create your own exception with name related to this situation
```suggestion
            throw new AccessDeniedException(
```